storage/tests/test_storage.py

```diff
@@ -19,37 +19,41 @@ class TestLibrary:
     def ensured_asset():
         """Creates an asset with a random name with a lifespan of the test case."""
         name = f"cosmin-{random_str(8)}"
-        asset = storage._create_asset(name)
-        yield asset
+        storage.set_text(name, random_str(16), wait=True)
+        yield name
         try:
             storage.delete_asset(name)
         except storage.AssetNotFound:
             pass
 
-    def test_list_assets(self, ensured_asset):
+    @staticmethod
+    @pytest.fixture(scope="class")
+    def cleanup_assets():
+        yield
         assets = storage.list_assets()
-        asset_names = [asset["name"] for asset in assets]
-        assert ensured_asset["name"] in asset_names
+        for asset in assets:
+            if asset.startswith("cosmin-"):
+                storage.delete_asset(asset)
 
-    def test_get_asset(self, ensured_asset):
-        value = storage.get_asset(ensured_asset["name"])
-        assert value == ""
+    def test_list_assets(self, ensured_asset):
+        assets = storage.list_assets()
+        assert ensured_asset in assets
 
     def test_delete_asset(self, ensured_asset):
-        name = ensured_asset["name"]
-        storage.delete_asset(name)
+        storage.delete_asset(ensured_asset)
         with pytest.raises(storage.AssetNotFound):
-            storage.delete_asset(name)
-
-    def test_set_asset(self):
-        name = "cosmin"
-        provided_value = "cosmin@robocorp.com"
-        storage.set_asset(name, provided_value)
-        retrieved_value = storage.get_asset(name)
-        assert retrieved_value == provided_value
-
-        # Setting also replaces already existing assets with newer values.
-        extra = " is my e-mail"
-        storage.set_asset(name, provided_value + extra)
-        retrieved_value = storage.get_asset(name)
-        assert retrieved_value == provided_value + extra
+            storage.delete_asset(ensured_asset)
+
+    def test_text_asset(self, cleanup_assets):
+        name = "cosmin-text"
+        text = "cosmin@robocorp.com"
+        storage.set_text(name, text)
+        obtained_text = storage.get_text(name)
+        assert obtained_text == text
+
+    def test_bytes_asset(self, cleanup_assets):
+        name = "cosmin - bytes"
```

Comment:
Currently this fails during later GET retrieval with: `FAILED tests/test_storage.py::TestLibrary::test_bytes_asset - robocorp.storage._client.AssetNotFound: 404 not_found - Asset with id 'name:cosmin - bytes' not found`

Reply:
The API is now ready to deal with spaces, but works with the `%20` encoding for spaces, therefore `quote` instead of `quote_plus` works here now. (cc @osrjv )